kyaml/fn/framework/command/example_test.go

```diff
@@ -310,7 +310,7 @@ func ExampleBuild_validate() {
 			}
 
 			// lookup replicas field
-			r, err := rl.Items[i].Pipe(yaml.Lookup("spec", "replicas"))
+			r, err := rl.Items[i].AsRNode().Pipe(yaml.Lookup("spec", "replicas"))
```

Comment:
Would it make sense to add `Pipe` as a KubeNode function so the developers who already use the old framework can smoothly upgrade and use the new KubeObj. For example, I have used the old `kyaml` lib to develop some KRM functions, and now I want to extend the functions but I do not want to change existing code when updating the lib.

Reply:
IMO `Pipe` is not very intuitive. And I don't plan to add it on KubeObject.
I'd suggest fn developers to use Get and Set to achieve their goal.

Reply:
Understood. My main concern is about the backward compatibility, especially since the `rl.Items[i].Pipe` is previously the key approach to call functions. 
Right now, it is imported by [35 public repos](https://pkg.go.dev/sigs.k8s.io/kustomize/kyaml/fn/framework?tab=importedby) (some are owned by us so it is fine), but repos like this https://github.com/kumorilabs/konvert/blob/917abd1eee5d6b0b234ca5efc0442377b8363187/internal/functions/functions.go#L56 would definitely be broken once updated. And the lib is also imported by other private repos.

Reply:
I'm not challenge the `Pipe` function. I'm wondering if it is safe to change the `Items` from `[]*Rnode` to `[]*KubeObject`. Do you think we can define another attribute in ResourceList and mark `Items` as `DEPRECATED`? 

Besides, do we plan to update `KubeObject` in the other repos owned by us or we have edit access? like kubebuilder, github.com/google/k8s-digester/cmd/function? @droot